dom.bs

```diff
@@ -3040,6 +3044,44 @@ standards that want to define APIs shared between <a for=/>documents</a> and
  <a for=CharacterData>data</a> is the string and <a for=Node>node document</a> is
  <var>document</var>.
 
+ <li><p>Let <var>isScriptElement</var> be true if <var>parent</var> is an {{HTMLScriptElement}}
+ <a for=/>node</a>; otherwise false.
+
+ <li>
+  <p><a for=set>For each</a> <var>value</var> of <var>nodes</var>:
+
+  <ol>
+   <li>
+    <p>If <var>value</var> is a <a for=/>node</a>, then:
+
+    <ol>
+     <li><p>If <var>isScriptElement</var> is false, then <a for=iteration>continue</a>.
+
+     <li><p>If <var>value</var> is not a {{Text}} <a for=/>node</a>, then <a for=iteration>continue</a>.
+
+     <li><p>Let <var>text</var> be the result of calling <a abstract-op>Get Trusted Type compliant string</a>,
+     with {{TrustedScript}}, <var>document</var>'s <a>relevant global object</a>,
+     <var>value</var>'s <a for=CharacterData>data</a>, "<code>HTMLScriptElement text</code>",
+     "<code>script</code>".
+
+     <li><p>Set <var>value</var>'s <a for=CharacterData>data</a> to <var>text</var>.
+    </ol>
+
+    <li>
+     <p>If <var>value</var> is a {{TrustedScript}}, then:
+
+     <ol>
+      <li><p>Let <var>text</var> be <var>value</var>'s <a for=TrustedScript>\[[Data]]</a>
+      internal slot value.
+
+      <li><p>Let <var>newValue</var> be a new {{Text}} <a for=/>node</a> whose
+      <a for=CharacterData>data</a> is <var>text</var> and <a for=Node>node document</a> is
+      <var>document</var>.
+
+      <li><p>Replace <var>value</var> with <var>newValue</var>.
```

Comment:
"Replace" is not defined here. I guess you are attempting to modify _nodes_ directly? That makes me worried about https://github.com/whatwg/infra/pull/451.

Reply:
Yeah, and tbf both WebKit and Chromium actually make a new list object that they append to, so perhaps that'd be better to spec like that? MIght also handle the fact that the trusted types check can do script execution a bit better too? (Though looping through a list and can potentially be modified might be a seperate issue?)

Reply:
That would sidestep this issue so that's probably better, yeah.

Reply:
Should this be a set or a list? Based on what infra says I *think* maybe a set? But then the nodes parameter comes from a JS union so is that maybe a list?

Reply:
I don't think it matters here and using list syntax is nice.